doc/user_guide/big_data.rst

```diff
@@ -0,0 +1,228 @@
+.. _big-data-label:
+
+Big data
+********
+
+.. versionadded:: 1.2
+
+This chapter describes the behaviour of :py:class:`~._signals.lazy.LazySignal`
+class and its derivatives.
+
+The LazySignal idea
+-------------------
+
+In usual (non-lazy) HyperSpy signals the actual data (accessed via the field
+``s.data``) is most often loaded into the computer memory for fast access and
+processing. While such behaviour indeed results in best performance, it
+obviously requires at least as much computer memory as the data, often twice
+that to store the results of any computation. This becomes a significant
+problem if processing large amounts of data on consumer-oriented hardware.
+
+HyperSpy offers a solution for this problem by including
+:py:class:`~._signals.lazy.LazySignal` and its derivatives. The main idea of
+these classes is to perform any operation (as the name suggests)
+`lazily <https://en.wikipedia.org/wiki/Lazy_evaluation>`_ (delaying the
+execution until the result is requested (e.g. saved, plotted)) and in a
+`blocked fashion <https://en.wikipedia.org/wiki/Block_matrix>`_. This is
+achieved by building a "history tree" (formally called a Directed Acyclic Graph
+(DAG)) of the computations, where the original data is at the root, and any
+further operations branch from it. Only when a certain branch result is
+requested, the way to the root is found and evaluated in the correct sequence
+on the correct blocks.
+
+The "magic" is performed by (for the sake of simplicity) storing the data not
+as ``numpy.ndarray``, but ``dask.array.Array`` (more information `here
+<https://dask.readthedocs.io/en/latest/>`_). ``dask`` offers a couple of
+advantages:
+
+* **Teoretically, arbitrary-sized data processing is possible**. By only
+  loading a couple of chunks at a time, theoretically any signal can be
+  processed, albeit slower. In practice, this may be limited: (i) some
+  operations may require certain chunking pattern, which may still saturate
+  memory; (ii) a couple of tens of chunks should fit into the computer memory
+  comfortably.
+* **Loading only the required data**. If a certain part (chunk) of the data is
+  not required for the final result, it will not be loaded at all, saving time
+  and resources.
+* **Able to extend to clusters**. ``dask.distributed`` (documentation `here
```

Comment:
Be clearer - "compute clusters"  or "a distributed computing environment" - clusters could mean k-means clustering...